plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonqFeatureDev/controller/FeatureDevController.kt

```diff
@@ -158,6 +164,32 @@ class FeatureDevController(
         }
     }
 
+    override suspend fun processChatItemFeedbackMessage(message: IncomingFeatureDevMessage.ChatItemFeedbackMessage) {
+        logger.debug { "$FEATURE_NAME: Processing ChatItemFeedbackMessage: ${message.comment}" }
+
+        val session = getSessionInfo(message.tabId)
+
+        val comment = FeedbackComment(
+            conversationId = session.conversationId,
+            userComment = message.comment.orEmpty(),
+            reason = message.selectedOption,
+            messageId = message.messageId,
+            type = "featuredev-chat-answer-feedback"
+        )
+
+        try {
+            TelemetryService.getInstance().sendFeedback(
+                sentiment = Sentiment.NEGATIVE,
+                comment = objectMapper.writeValueAsString(comment),
+            )
+            logger.info { "$FEATURE_NAME answer feedback sent: \"Negative\"" }
+        } catch (e: Throwable) {
+            e.notifyError(message("feedback.submit_failed", e))
+            logger.warn(e) { "Failed to submit feedback" }
+            return
+        }
+    }
+
```

Comment:
not happy with how this is a copy paste from the regular chat impl

Reply:
Shall I move the common code to a common place as a helper, or modify TelemetryService.sendFeedback to take feature type?

Reply:
the former makes more sense